packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2024,13 +2063,33 @@ export class KeyringController extends BaseController<
       releaseLock: MutexInterface.Releaser;
     }) => Promise<T>,
   ): Promise<T> {
-    const releaseLock = await this.#vaultOperationMutex.acquire();
-
-    try {
-      return await fn({ releaseLock });
-    } finally {
-      releaseLock();
+    if (!this.#controllerOperationMutex.isLocked()) {
+      throw new Error(KeyringControllerError.ControllerLockRequired);
     }
```

Comment:
This helps enforcing the usage of locks in _all_ public methods that need to persist